packages/sources/ipfs/src/endpoint/read.ts

```diff
@@ -0,0 +1,82 @@
+import { Requester, Validator } from '@chainlink/ea-bootstrap'
+import { ExecuteWithConfig, Config } from '@chainlink/types'
+import { create } from 'ipfs-http-client'
+import { deserialize } from '../codec'
+import { IPFS } from 'ipfs-core-types'
+import { CID } from 'multiformats/cid'
+
+export type IPFSPath = string | CID
+export { CID }
+
+export const supportedEndpoints = ['read']
+
+const customParams = {
+  cid: false,
+  ipns: false,
+  codec: false,
+  type: false,
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
+  const validator = new Validator(request, customParams)
+  if (validator.error) throw validator.error
+
+  const jobRunID = validator.validated.id
+  let cid: IPFSPath = validator.validated.data.cid
+  const ipns = validator.validated.data.ipns
+  const type = validator.validated.data.type || 'raw'
+
+  if (!cid && !ipns) {
+    throw Error('Request is missing both "cid" and "ipns". One is required.')
+  }
+
+  const client = create({ url: config.api.baseURL })
+
+  // If CID is not included, we try to resolve IPNS
+  if (!cid) {
+    for await (const ipfsCid of client.name.resolve(`/ipns/${ipns}`)) {
+      cid = ipfsCid.replace('/ipfs/', '')
+    }
+  }
+
+  const codec = validator.validated.data.codec
+
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  let result: string | object
```

Comment:
Could we use something like 
```
{ [key: string]: unknown }
```
on these unknown object types?

@HenryNguyen5 do you know of a better way TypeScript master?

Reply:
Looks like the `object` type is being propagated from the ipfs library, to make it more apparent i've suggested changed via using referenced types in https://github.com/smartcontractkit/external-adapters-js/pull/921